src/Http/Http.Features/src/FeatureCollection.cs

```diff
@@ -73,7 +89,7 @@ public virtual int Revision
 
                 if (_features == null)
                 {
-                    _features = new Dictionary<Type, object>();
+                    _features = new Dictionary<Type, object>(_initialCapacity);
```

Comment:
This means that in the default case we're now passing in 0, even though we know we're about to add at least one item. How well does Dictionary handle that?

Reply:
Last I checked dictionary starts at 0 either way and only after the first add allocates space to the next nearest prime from a predefined list, which is 3, then 7 and so on.

Reply:
[Dictionary's default capacity is 0](https://github.com/dotnet/runtime/blob/82ca681cbac89d813a3ce397e0c665e6c051ed67/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/Dictionary.cs#L39-L45)